app/modules/ads/tracking/postmessage-tracker.js

```diff
@@ -0,0 +1,41 @@
+import { track, TrackingMethod } from '../../../utils/track';
+
+export function registerPostmessageTrackingTracker() {
+  const {
+    PostmessageTracker,
+    TrackingTarget,
+    trackingPayloadValidationMiddleware,
+  } = window.Wikia.adEngine;
+
+  const postmessageTracker = new PostmessageTracker(
+    ['payload', 'target'],
+  );
+
+  return postmessageTracker
+    .add(trackingPayloadValidationMiddleware)
+    .register(
+      (message) => {
+        const { target, payload } = message;
+
+        switch (target) {
+          case TrackingTarget.GoogleAnalytics: {
+            track(Object.assign({}, payload, {
```

Comment:
why not spread?

Reply:
Spread is not supported in this place. Adding it results in an error.